### PR TITLE
ENH More standardisation

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Description

Two commits:
1. The bulk of the standardisation - removing some more unused code, adding a test, etc.
2. Just moving some methods around (and some very minor changes to phpdoc). I did this separately so it'll be easier to review the actual changes being made.

## Issues

- https://github.com/silverstripe/silverstripe-linkfield/issues/78

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

File: src/Models/FileLink.php [hunk 1]
```diff
@@ -52,10 +52,10 @@ public function getURL(): string
         return $file->exists() ? (string) $file->getURL() : '';
     }
 
-    public function getDefaultTitle(): string
+    protected function getDefaultTitle(): string
```
Comment (ChloeSartorelli) [inline src/Models/FileLink.php]: Never called outside of `Link` models except in tests.
Everywhere else should use `getTitle()` to get the real title.

File: src/Models/FileLink.php [hunk 2]
```diff
@@ -52,10 +52,10 @@ public function getURL(): string
...
     {
         $file = $this->File();
-        if (!$file->exists()) {
+        if (!$file?->exists()) {
```
Comment (ChloeSartorelli) [inline src/Models/FileLink.php]: Matches how it's used elsewhere in this class.

File: src/Models/Link.php [hunk 3]
```diff
@@ -2,10 +2,8 @@
 
 namespace SilverStripe\LinkField\Models;
 
-use InvalidArgumentException;
 use ReflectionException;
 use SilverStripe\Core\ClassInfo;
-use SilverStripe\Core\Injector\Injector;
```
Comment (ChloeSartorelli) [inline src/Models/Link.php]: Not used anymore

File: src/Models/Link.php [hunk 4]
```diff
@@ -69,17 +67,20 @@ class Link extends DataObject
...
+     */
     public function getDescription(): string
     {
         return '';
     }
 
-    public function scaffoldLinkFields(array $data): FieldList
-    {
-        return $this->getCMSFields();
-    }
```
Comment (ChloeSartorelli) [inline src/Models/Link.php:80]: Unused code

File: src/Models/Link.php [hunk 5]
```diff
@@ -69,17 +67,20 @@ class Link extends DataObject
...
-
-    public function LinkTypeHandlerName(): string
+    /**
+     * Get the react component used to render the modal form.
+     */
+    public function getLinkTypeHandlerName(): string
```
Comment (ChloeSartorelli) [inline src/Models/Link.php]: Getters should be prefixed with `get`

File: src/Models/Link.php [hunk 6]
```diff
@@ -132,91 +133,9 @@ public function onBeforeWrite(): void
         parent::onBeforeWrite();
     }
 
-    function setData($data): Link
```
Comment (ChloeSartorelli) [inline src/Models/Link.php:135]: Unused method

File: src/Models/Link.php [hunk 7]
```diff
@@ -237,19 +156,6 @@ public function jsonSerialize(): mixed
         return $data;
     }
 
-    public function loadLinkData(array $data): Link
```
Comment (ChloeSartorelli) [inline src/Models/Link.php:240]: Unused method

File: tests/php/Models/FileLinkTest.php [hunk 8]
```diff
@@ -18,7 +19,7 @@ class FileLinkTest extends SapphireTest
     public function testGetDescription(): void
     {
         // FileLink without a page
-        $link = FileLink::create();
+        $link = new FileLink();
```
Comment (ChloeSartorelli) [inline tests/php/Models/FileLinkTest.php:22]: Tests should avoid using injector when they're not explicitly testing injector

File: tests/php/Models/FileLinkTest.php [hunk 9]
```diff
@@ -35,4 +36,26 @@ public function testGetDescription(): void
         $link->write();
         $this->assertSame('file-b.png', $link->getDescription());
     }
+
+    public function testGetDefaultTitle(): void
```
Comment (ChloeSartorelli) [inline tests/php/Models/FileLinkTest.php:40]: Matches a test in `SiteTreeLinkTest`

**Review by emteknetnz — APPROVED** [review]
Code change look fine, have smoke tested locally

**Pull request merged**